src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/TasksTest.kt

```diff
@@ -858,9 +860,26 @@ class TasksTest : IntegrationTestBase() {
 
       @ParameterizedTest
       @CsvSource(
-        "assessment,WOMENS", "assessment,PIPE", "assessment,ESAP", "assessment,EMERGENCY",
-        "placementRequest,WOMENS", "placementRequest,PIPE", "placementRequest,ESAP", "placementRequest,EMERGENCY",
-        "placementApplication,WOMENS", "placementApplication,PIPE", "placementApplication,ESAP", "placementApplication,EMERGENCY",
+        "assessment,WOMENS",
+        "assessment,PIPE",
+        "assessment,ESAP",
+        "assessment,EMERGENCY",
+        "assessment,RECOVERY_FOCUSED",
+        "assessment,MENTAL_HEALTH_SPECIALIST",
+
+        "placementRequest,WOMENS",
+        "placementRequest,PIPE",
+        "placementRequest,ESAP",
+        "placementRequest,EMERGENCY",
+        "placementRequest,RECOVERY_FOCUSED",
+        "placementRequest,MENTAL_HEALTH_SPECIALIST",
+
+        "placementApplication,WOMENS",
+        "placementApplication,PIPE",
+        "placementApplication,ESAP",
+        "placementApplication,EMERGENCY",
+        "placementApplication,RECOVERY_FOCUSED",
+        "placementApplication,MENTAL_HEALTH_SPECIALIST",
       )
       fun `Get all tasks filters by task type and required qualification`(taskType: TaskType, qualification: UserQualification) {
         val url = "/tasks?type=${taskType.value}&requiredQualification=${qualification.name.lowercase()}"
```

Comment:
are there corresponding tasks in the tasks collection for the new types?

I can see at line 820 (in the old code) something like this...

              listOf(
                UserQualification.WOMENS,
                UserQualification.ESAP,
                UserQualification.PIPE,
              ).forEach { qualification ->

Do we need to add to this? Would it be good to add an assertion at line 887 to make sure expected tasks is not empty? Or am i misreading this :)

Reply:
👍 yes I think is covered in https://github.com/ministryofjustice/approved-premises-api/blob/e343d80d274904f6dc6bdb9f6e50ba8fc6421eed/src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/TasksTest.kt#L820-L900

There's a bunch of iteration which builds up to define the `tasks` fixture:

```kt
    tasks = mapOf(
                TaskType.assessment to assessmentTasks,
                TaskType.placementRequest to placementRequestTasks,
                TaskType.placementApplication to placementApplicationTasks,
              )
```
and the parameterised tests then assert against elements of that tasks fixture, e.g. `tasks['placementRequest']['RECOVERY_FOCUSED']`

Probably we could make this easier to read!

Reply:
Perfect, i completely missed that. Thanks.